app/features/receive/lightning-address-service.tsx

```diff
@@ -0,0 +1,212 @@
+import { getCashuWallet } from '~/lib/cashu';
+import type {
+  LNURLError,
+  LNURLPayParams,
+  LNURLPayResult,
+  LNURLVerifyResult,
+} from '~/lib/lnurl/types';
+import { Money } from '~/lib/money';
+import { AccountRepository } from '../accounts/account-repository';
+import type { BoardwalkDb } from '../boardwalk-db/database';
+import type { CashuCryptography } from '../shared/cashu';
+import { UserRepository } from '../user/user-repository';
+import { CashuReceiveQuoteRepository } from './cashu-receive-quote-repository';
+import { CashuReceiveQuoteService } from './cashu-receive-quote-service';
+
+// TODO: Is there a better way to do this? We don't need any of this for the lightning address service to work,
+// but there could be some hidden bugs that appear if we do it this way. We could also make each of these optional
+// and only set the ones we need, then throw an error in the services and repositories if the needed methods are not set.
+const fakeCryptography: CashuCryptography = {
+  encrypt: async <T = unknown>(data: T) => JSON.stringify(data),
+  decrypt: async <T = unknown>(data: string) => data as T,
+  getSeed: () => Promise.resolve(new Uint8Array()),
+  getXpub: () => Promise.resolve(''),
+  signMessage: () => Promise.resolve(''),
+};
+
+const getOriginFromRequest = (request: Request) => {
+  const url = new URL(request.url);
+  const protocol = url.origin.includes('localhost') ? 'http' : url.protocol;
```

Comment:
why not keep whatever protocol is used? so if localhost uses https keep it and if uses http also keep it